smart-server/src/main/java/org/smartdata/server/actions/ActionRegister.java

```diff
@@ -0,0 +1,94 @@
+package org.smartdata.server.actions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smartdata.common.actions.ActionType;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+/**
+ * Provide Action Map and Action instance
+ */
+public class ActionRegister {
+  static final Logger LOG = LoggerFactory.getLogger(ActionRegister.class);
+
+  private static ActionRegister instance = new ActionRegister();
+  HashMap<String, Class> actionMap;
+
+  synchronized public static ActionRegister getInstance() {
+    return instance;
+  }
+
+  private ActionRegister() {
+    actionMap = new HashMap<>();
+  }
+
+  public void initial() {
+    loadNativeAction();
+    loadUserDefinedAction();
+  }
+
+  public void loadNativeAction() {
+    String path = "org.smartdata.server.actions.";
+    for (ActionType t : ActionType.values()) {
+      String name = t.toString();
+      try {
+        actionMap.put(name, Class.forName(path + name));
+      } catch (ClassNotFoundException e) {
+        LOG.info("Class {} not found!", name);
+        continue;
+        // actionMap.put(name, null);
+      }
+    }
+  }
+
+  public void loadUserDefinedAction() {
+    // TODO Replace with Dir
+    String pathToJar = "/home/intel/Develop/SSM/Smart/User/UDAction.jar";
```

Comment:
After we adding configuration and SSM context, this part will be configureable (rather than hardcoded). Currently, I think we should focus on CommandSubmit and getCommandFromDB. This two paired methods are related with Action generation and consumption.